### PR TITLE
Fix purge AWS resources workflow

### LINKED ISSUE #1702: Purge AWS EKS Clusters not purging any clusters
It looks like the https://github.com/radius-project/samples/actions/workflows/purge-aws-eks-clusters.yaml action isn't purging old clusters.

https://github.com/radius-project/samples/actions/runs/10365172732/job/28691813887

```
An error occurred (InvalidParameterException) when calling the DescribeCluster operation: The parameter name should not be greater than 100 characters.
```

[AB#12984](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/12984)

### PR DESCRIPTION
**Pull request by willdavsmith**
Sometimes the "Purge AWS EKS Clusters" workflow will fail and not delete any eks clusters. [Example](https://github.com/radius-project/samples/actions/runs/10954779362/job/30417354221). This will cause the samples tests to fail because of resource quotas for NAT Gateways, Elastic IP Addresses, and so on. This PR adds some logic to address the issue as well as fail the workflow on script failure for better reporting.

Fixes: https://github.com/radius-project/samples/issues/1702

### CODE REVIEW

**Review by ytimocin — APPROVED** [review]

**Review by kachawla — COMMENTED** [review]
Could you summarize the fix in the PR description? Just good practice to enable better PR reviews.

**Review by ytimocin — APPROVED** [review]
